packages/node/src/evm/templates/template-fetching.test.ts

```diff
@@ -175,28 +173,182 @@ describe('fetch (templates)', () => {
     expect(getTemplatesMock).toHaveBeenCalledTimes(2);
   });
 
-  it('retries a maximum of two times', async () => {
-    const rawTemplates = {
-      designatedWallets: ['designatedWallet-0'],
-      endpointIds: ['endpointId-0'],
-      providerIds: ['providerId-0'],
-      fulfillAddresses: ['fulfillAddresses-0'],
-      fulfillFunctionIds: ['fulfillFunctionId-0'],
-      parameters: ['0x6874656d706c6174656576616c7565'],
-      requesterIndices: ['requesterIndex-0'],
+  it('fetches templates individually if the template group cannot be fetched', async () => {
+    getTemplatesMock.mockRejectedValueOnce(new Error('Server says no'));
```

Comment:
`OPERATION_RETRIES` is hardcoded in all these tests. Probably not worth it to fix right now but we can create an issue.

Reply:
Not sure I understand?

Reply:
No, I'm saying changing the `2` here
https://github.com/api3dao/airnode/blob/7472b30861d442602ea6176ad7a6a628de6a0e5a/packages/node/src/constants.ts#L32
would result in the tests failing. I think the point of keeping the parameters in one file is that you would be able to change the value in that file and wouldn't have to do anything else.

Reply:
It shouldn't break any tests as functionality would be the same. So these changes:

1. Change the `OPERATION_RETRIES` value to be `1`
2. Change the internal parts of the `retryOperation` function to work with values that are one less.

The end result would be:

```ts
// Good: Retry once on failure
retryOperation(1, () => myFunction());

// Bad: implies 2 retries
retryOperation(2, () => myFunction());
```

Reply:
I meant all the mocks in the tests assume `OPERATION_RETRIES` is 2

Reply:
Updated 2->3, 24 tests fail. I created the issue, we can deal with it later.

Reply:
Yeah most mocks only expect a single retry so that makes sense.

Going from 2 -> 1 wouldn't break anything (but we need to update the internals of the `retryOperation()` function too)

Reply:
Ah I see what you mean now. I wasn't referring to the TODO about changing 2 to 1 at all. What I'm saying is that say someone wants the node to retry one more time and they increment `OPERATION_RETRIES` at `constants.ts`. They would also have to update all the tests. Similarly if they didn't want any retries, they would have to decrement that variable and also update all the tests. Ideally the tests would import `OPERATION_RETRIES` and create the mocks with a loop.

Reply:
Yeah I figured we were talking about different things. With some of the tests we could make the following changes which would work regardless of how many times the mock is called

```js
mockReturnValueOnce() -> mockReturnValue()
mockResolvedValueOnce() -> mockResolvedValue()
mockRejectedValueOnce() -> mockRejectedValue()
```

I didn't do this initially because I preferred more fine grained control over what gets mocked, but it's not a big deal to switch. This wouldn't work for tests where the initial call fails and the subsequent call succeeds.

Reply:
I actually like the fine grained mocks. I think if we ever make this change
```js
getTemplatesMock.mockRejectedValueOnce(new Error('Server says no'));
getTemplatesMock.mockResolvedValueOnce(rawTemplates);
```
should be replaced with
```js
for (var i = 0; i < OPERATION_RETRIES - 1; i++) {
  getTemplatesMock.mockRejectedValueOnce(new Error('Server says no'));
}
getTemplatesMock.mockResolvedValueOnce(rawTemplates);
```

Reply:
Yeah something like that could work